### PR TITLE
Mac OSX bootstrap: Add dialog dependency and homebrew installation

### PR DESCRIPTION
**Pull request by BKreisel**
Add dialog dependency identified in #413 and install homebrew for user if not currently installed

### CODE REVIEW

File: bootstrap/mac.sh [hunk 1]
```diff
@@ -1,2 +1,10 @@
 #!/bin/sh
+if hash brew 2>/dev/null; then
+    echo "Homebrew Installed"
```
Comment (kuba) [inline bootstrap/mac.sh]: I'd skip the message :)

File: bootstrap/mac.sh [hunk 2]
```diff
@@ -1,2 +1,10 @@
...
+else
+    echo "Homebrew Not Installed\nDownloading..."
+    ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
```
Comment (kuba) [inline bootstrap/mac.sh]: Oh wow, looks a bit scary + seems strange to download `master`, but this is what they suggest on http://brew.sh/.

**Comment by kuba** [comment]
LGTM

**Comment by kuba** [comment]
By the way, welcome to the project and thanks for the contribution! :)

Would you like to take over #724?

**Comment by BKreisel** [comment]
I admit, the Homebrew download looks a bit terrifying, but apparently that is their official process. I'd be more than happy to help with Mac contributions and wouldn't mind taking over #724.

**Comment by BKreisel** [comment]
Thanks for the warm welcome kuba

**Comment by pde** [comment]
The "master" thing is indeed terrifying, but it's possibly a reflection of unsafe practices throughout the project rather than an isolated thing we could ask them to fix quickly...

**Comment by DomT4** [comment]
:wave: What's the worry/confusion over Homebrew here?

**Comment by jmhodges** [comment]
I do not use homebrew and do not want letsencrypt installing it. Please remove that code.

**Comment by jmhodges** [comment]
Sigh, I'm an idiot. I misread what was going on here. Sorry.

**Comment by BKreisel** [comment]
Jeff,

What was the confusion?

Brandon

Brandon

**Pull request merged**